### PR TITLE
Fix: must take xlct into account while determine_rtt

### PR DESCRIPTION
**Pull request by wangfuyu**
Problem:
Client request with 0RTT paras may failed with timeout but not success with 1RTT.

Reproduce:
1) http_client launch 0rtt requests with -0 prog_0rtt.info, like
http_client -H test.com -s 127.0.0.1:3333 -p / -o version=Q043 -n 1 -r 10 -R 1 -0 ./prog_0rtt.info
2) server update CERT
3) http_client launch 0rtt again
4) http_client failed with timeout but not success with 1RTT.

Analysis:
 Lsquic ignore XLCT tag while parse_hs_data and not cmp XLCT with hash(enc_session->cert_ptr)

### CODE REVIEW

**Comment by dtikhonov** [comment]
Thank you!  We will examine this shortly.

File: src/liblsquic/lsquic_handshake.c [hunk 1]
```diff
@@ -2346,6 +2364,9 @@ handle_chlo_frames_data(const uint8_t *data, int len,
         break;
     }
 
+    rtt = determine_rtts(enc_session, ip, t);
```
Comment (dtikhonov) [inline src/liblsquic/lsquic_handshake.c:2367]: Moving of this code, while an improvement, does not seem related to the fix?

Comment (dtikhonov) [reply src/liblsquic/lsquic_handshake.c:2367]: I take it back! :-)

Comment (wangfuyu) [reply src/liblsquic/lsquic_handshake.c:2367]: > I take it back! :-)

You must not take it back, because enc_session->cert_ptr is set after get_sni_SSL_CTX.
If you move it back, will coredump.

Comment (dtikhonov) [reply src/liblsquic/lsquic_handshake.c:2367]: By "take back," I meant that I retract my original comment.

**Comment by dtikhonov** [comment]
I understand what the patch does, but I am not able to reproduce the symptom where the client times out.  In Step (4) above, do you mean that one of the 0-RTT connections issues in Step (3) fails?

Which server are you using?

**Comment by dtikhonov** [comment]
> I am not able to reproduce the symptom where the client times out.

I can reproduce it now.
